### ISSUE TITLE
"Contributing" section of README.rst tells users to pip install, should be pip3 install

### ISSUE THREAD

**Issue by DeeDeeG** [trufont/trufont#408]
Step 6 in the contributing part of README.rst says to `pip install --editable .` which doesn't work.

`pip3 install --editable .` does work, however.

Also a minor thing: steps are 1,**2,4**...**9** should be 1,**2,3**...**8**

**Comment by eliheuer** [comment]
Are you working in a venv? Maybe it would be a good idea to make it clear in the readme that you should be in a venv when working in "editable" mode. 

I'm on Ubuntu 16.10 and if I do this `python3 -m venv tf-venv` 

`pip --version` gives me this:

`pip 9.0.1 from /home/elijah/Projects/tf-venv/lib/python3.5/site-packages (python 3.5)`

**Comment by DeeDeeG** [comment]
>Maybe it would be a good idea to make it clear in the readme that you should be in a venv when working in "editable" mode.

Yeah, I think so. To put it another way; if that's implied in the current instructions, I think it would be better to say it than imply it.

(For some context, I've been through this topic here: https://github.com/trufont/trufont/issues/404, and thought I had it all figured out, that the only problem was the pip install, vs pip3 install. Maybe the venv is also needed to make the install work? It feels like there are a few too many variables, but I have been trying to sort them out.)

**Comment by adrientetar** [comment]
You don't have to be in a virtual environment to install a package in editable mode (I don't see why you would?). I'm not, for instance.

**Comment by anthrotype** [comment]
The stuff about using virtual environment in the README.md is just a recommendation, to avoid clashes and version mismatch with other packages in your global site-packages, given TruFont's long and deep list of dependencies. If you prefer to pip install --user or even sudo pip install, there's no one stopping you. Though you've been warned ;)

**Comment by DeeDeeG** [comment]
Okay, so regarding the title of this issue:

On Ubuntu 16.04... TruFont does not install using `pip install -e`. It does install using `pip3 install -e`.

(from the github/master branch)

The error message isn't very obvious either.

So for ubuntu users such as myself, having the instructions tell them: "use pip3", would be helpful.

**Comment by anthrotype** [comment]
Trufont works only with python3.
The global `pip` command is linked to the default `python`, which in most environments is equivalent to `python2`; `pip3` is the pip script that is linked to `python3`.
Inside a venv, `pip` is linked to the virtual `python` interpreter, which could be either 2 or 3 depending on the global python interpreter used to create the virtual environment.
The `pip3` command is also available from inside a python3's virtual environment, so I guess the README can be changed to use `pip3` to make it explicit that you must use the python3's pip and not any pip.
Maybe we could also add `python_requires` keyword to setup.py to prevent users from accidentally installing TruFont on python 2, but that only works with very recent versions of setuptools and pip.

**Comment by adrientetar** [comment]
I do have the pip command on my 16.04 image, but I think I didn't install it from apt as it's way too old in there. I try not install python packages from apt, I just do easy_install pip then use pip for everything. Anyway, maybe this is something to take to your distro maintainers.

And there's no python3 in WIndows, but the README is using it. Maybe we should say to use python on Windows or python3 on other platforms and then python3 -m pip.

**Comment by DeeDeeG** [comment]
I do know what seems to be the sane way of getting this to run on recent Ubuntu distros. It's the same as the contributing instructions, other than getting around the lack of brotli from pypi. (I am not including steps about git branch management and such, just getting it to launch.)

0. Clone the repo (`git clone https://github.com/trufont/trufont`)
1. Make a venv (`python3 -m venv VENV_DIR`) and activate (`source VENV_DIR/bin/activate`)
2. Edit requirements.txt & setup.py to delete the line mentioning brotli. removes brotli dependency... OPTIONAL: compile and install brotli, used for exporting to WOFF2 only
3. `pip3 install --editable .`

How to launch, troubleshooting:

4. (run trufont by typing `trufont`)
5. (if there are permissions errors trying to write to *python-so-and-so*.egg, you probably ran sudo, which [I now know] is a bad idea. Wipe the trufont folder and re-clone. Easier than `chown` and `chmod`.)

So out of that, my suggestions would be to mention how to remove the brotli dependency (for Linux users), mention `pip3 install` in case users on Linux/Mac/etc. decide not to install in a venv, and possibly explain that the permissions errors can be fixed by wiping and re-cloning trufont, and don't try to fix permissions errors by running `sudo pip install`.

**Comment by DeeDeeG** [comment]
> there's no python3 in Windows,

I believe there is, I have had to download it for other projects.

![python3 windows](https://cloud.githubusercontent.com/assets/20157115/24831977/45654aa6-1c73-11e7-8ea9-6271115e8861.png)

**Comment by adrientetar** [comment]
> I believe there is, I have had to download it for other projects.

I don't disagree, I was saying that there's no "python3" command in the terminal on Windows (at least by default) like you have on OSX/Linux, it's only "python".

**Comment by DeeDeeG** [comment]
I just tried to install on Windows Vista (with 32-bit python 3.6.1), but ran into some issues... I don't really mind the issues though, since I'm just trying to find out what the right install steps are for various platforms.

Here were my steps:

- installed python 3.6.1

- ran `python -m venv VENV`

- ran `VENV\Scripts\activate.bat` (all steps after this done in the venv)

- cloned the repo from GitHub

- `pip install --editable  .` fails ("Access is denied")

- `python -m pip -- version` gives me: 9.0.1

- `python -m pip install --editable .` failed, with an error having to do with brotli

- had to delete brotli requirement... running again...

 error message this time:

>This application failed to start because it could not find or load the Qt platform plugin "windows"
> in "".
>
>Reinstalling the application may fix this problem.

I don't really need to *use* the program on Windows, so I don't mind the error messages. Just wanted to know what commands are needed for installing.. Particularly: `python -m pip install -e .`; I was able to confirm that running `python3` doesn't do anything on Windows. It just says command not found.

I might try to write up a PR for updating the readme once/if I can test how to install it properly on Mac.

**Comment by adrientetar** [comment]
Did you install the requirements? pip install -r requirements.txt  
Check that import PyQt5 is working.

I don't know about the "access denied" thing you got, I never had that on Windows. Maybe related to venv since I don't use that.

**Comment by anthrotype** [comment]
inside the venv, `pip --version` should produce the same result as `python -m pip --version`, i.e. pip should be linked to the virtual python executable, and you should not experience any differences if you run one or the other. Please make sure once again that is the case.

Do you have any other python in your %PATH%? Do `echo %PATH%`, and `where python` and check that the Python you think you are using is the one you are actually using. You may also use the `py` [Python launcher](https://docs.python.org/3/using/windows.html#launcher) in case you have multiple pythons on Windows, so instead of calling just `python` or `pip`, you would use `py -3.6`, and `py -3.6 -m pip ...`

Brotli, yes there's no official 3.6 wheels yet (i'll work on that soonish.. btw, they recently tagged a new v0.6.0). So to compile you need a compiler. For Windows and python 3.6, you can install MS Build Tools 2015:
https://www.microsoft.com/en-us/download/details.aspx?id=48159

That error about Qt failing to load the plaform plugins is a bit suspicious.. I'll have to check on my local windows machine to see if they broke something in one of the latest PyQt5 releases. I remember that in my previous tests, PyQt5 was working fine inside a `python3 -m venv` kind of virtual environment (the native virtual environments of python3), but failed to load some DLLs if was installed inside a `virtualenv` kind of virtual environment (the third-party package that you pip install virtualenv).

**Comment by DeeDeeG** [comment]
I do have folders called "Python26" and "Python27" in my C:\ directory, but I don't remember if I ever installed them.

I ran some commands you mentioned. First one is `pip --version`, outside of any venv. Next command is `where python`. Last command is echo `%PATH%`..

![cmd](https://cloud.githubusercontent.com/assets/20157115/25094894/1de4addc-2367-11e7-9ea6-75a28b5444cd.png)

**Comment by DeeDeeG** [comment]
Worked much better for me on Windows 7.

Steps:

- have python 3.6 already installed

- clone the repo.

- make a venv, and activate. (`python -m venv VENV`; `VENV\Scripts\activate.bat`)

- delete brotli requirement from requirements.txt and setup.py 

- cd into trufont dir, `pip install -e .`

- type `trufont` to launch (launches without errors)

**Comment by anthrotype** [comment]
Yes, I meant that one.

**Comment by adrientetar** [comment]
I'll remove brotli since it doesn't have wheels.

**Comment by anthrotype** [comment]
Sgtm. We can readd once wheels for 3.6 are available.

**Comment by anthrotype** [comment]
Brotli has wheels for 3.6 now:
https://pypi.python.org/pypi/Brotli/0.6.0
